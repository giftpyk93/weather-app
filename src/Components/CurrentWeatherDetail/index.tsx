import React, { FC } from "react";
import Grid from "@material-ui/core/Grid";

type WeatherDetailType = {
    label: string;
    value: string;
}

type CurrentWeatherDetailPropsType = {
    weatherDetails: WeatherDetailType[]
};

const CurrentWeatherDetail: FC<CurrentWeatherDetailPropsType> = ({ weatherDetails }) => {
  return (
    <Grid container className="d-flex flex-column p-3">
      <Grid item>
        <h6>CURRENT DETAIL</h6>
      </Grid>
      {weatherDetails.map((weatherDetail: WeatherDetailType) => {
          const {label, value} = weatherDetail
        return (
          <Grid key={label} item container xs={12} className="d-flex flex-row">
            <Grid item xs={1} />
            <Grid item xs={5} className="pl-4">
              {label}
            </Grid>
            <Grid item xs={1} />
            <Grid item xs={5} className="pl-4">
              {value}
            </Grid>
          </Grid>
        );
      })}
    </Grid>
  );
};

export default CurrentWeatherDetail;
