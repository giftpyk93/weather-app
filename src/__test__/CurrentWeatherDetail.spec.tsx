import React from "react";
import { create } from "react-test-renderer";
import CurrentWeatherDetail from "Components/CurrentWeatherDetail";

const mockWeatherDetail = [
  {
    label: "Humidity",
    value: "56%",
  },
  {
    label: "Wind",
    value: "15 meter/sec",
  },
  {
    label: "Pressure",
    value: "110 hPa",
  },
];

describe("CurrentWeatherDetail component", () => {
  test("Matches the snapshot", () => {
    const renderer = create(<CurrentWeatherDetail weatherDetails={mockWeatherDetail} />);
    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
