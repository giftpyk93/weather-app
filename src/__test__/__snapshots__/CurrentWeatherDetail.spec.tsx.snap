// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentWeatherDetail component Matches the snapshot 1`] = `
<div
  className="MuiGrid-root p-3 MuiGrid-container MuiGrid-direction-xs-column"
>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <h6>
      CURRENT DETAIL
    </h6>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
    />
    <div
      className="MuiGrid-root pl-4 MuiGrid-item MuiGrid-grid-xs-5"
    >
      Humidity
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
    />
    <div
      className="MuiGrid-root pl-4 MuiGrid-item MuiGrid-grid-xs-5"
    >
      56%
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
    />
    <div
      className="MuiGrid-root pl-4 MuiGrid-item MuiGrid-grid-xs-5"
    >
      Wind
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
    />
    <div
      className="MuiGrid-root pl-4 MuiGrid-item MuiGrid-grid-xs-5"
    >
      15 meter/sec
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
    />
    <div
      className="MuiGrid-root pl-4 MuiGrid-item MuiGrid-grid-xs-5"
    >
      Pressure
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
    />
    <div
      className="MuiGrid-root pl-4 MuiGrid-item MuiGrid-grid-xs-5"
    >
      110 hPa
    </div>
  </div>
</div>
`;
